openmcyclus/DepleteReactor.py

```diff
@@ -8,8 +11,525 @@ class DepleteReactor(Facility):
+        exist there already. 
+
+        If the reactor has a finite lifetime, calculate the number of 
+        cycles left and the number of assemblies needed for the 
+        lifetime of the reactor. Order whichever number is 
+        lower. 
+        
+        If the reactor does not need more fuel or is retired, then 
+        submit no bids for materials.
+        '''
+        port = []
+        n_assem_order = self.n_assem_core - self.core.count + self.n_assem_fresh + self.fresh_fuel.count
+        if self.exit_time != -1:
+            time_left = self.exit_time - self.context.time + 1
+            time_left_cycle = self.cycle_time + self.refuel_time - self.cycle_step
+            n_cycles_left = (time_left - time_left_cycle)/(self.cycle_time + self.refuel_time)
+            n_cycles_left = math.ceil(n_cycles_left)
+            n_need = max(0, n_cycles_left*self.n_assem_batch - self.n_assem_fresh + self.n_assem_core - self.core.count)
+            n_assem_order = min(n_assem_order, n_need)
+
+        if n_assem_order == 0:
+            return port
+        elif self.retired():
+            return port
+
+        for ii in range(n_assem_order):
+            for commod in self.fuel_incommods:
+                recipe = self.context.get_recipe(commod)
+                material = ts.Material.create_untracked(self.assem_size, recipe)
+            lib.record_time_series("demand"+commod, self, self.assem_size)
+            port.append({"commodities":{commod:material}, "constraints":self.assem_size})
+        return port
+
+    def get_material_bids(self, requests): # phase 2
+        '''
+        Read bids for fuel_outcommods and return bid portfolio.
+
+        If the unique_commods_ string is empty, add the names of 
+        the fuel out commodities.
+
+        For each of the items in the unique commodities list, 
+        if there are no requests for a given commodity name, then 
+        continue to the next commodity. Get the recipe for 
+        each commodity requested.
+
+        Looking at the composition for each commodity, 
+        if there is no composition, then move to the next commodity. 
+
+        For each request of each commodity, sum the total request 
+        for the commodity. Then create a bid for each request of 
+        each commodity.
+
+        For each material composition sum to total mass of 
+        the composition.
+
+        Add a constraint of the total quantity of the commodity available.
+        Create a bid portfolio for each request that can be met. 
+        '''
+        got_mats = False
+        bids = []
+        port = []
+        for commod in self.fuel_outcommods:
+            reqs = requests[commod]
+            if len(reqs) == 0:
+                continue
+            elif (got_mats == False):
+                all_mats = self.peek_spent()
+
+            if len(all_mats) == 0:
+                tot_qty = 0
+                continue
```

Comment:
I think for the logic of this method, you need the continue in there. Because if you don't have it, then it will proceed to the other parts of the for loop and run into the issue of there is nothing in the dictionary `all_mats` for it to do things with.

Reply:
When I removed the continue line, the tests [didn't pass](https://github.com/arfc/openmcyclus/actions/runs/4246424628)

Reply:
`continue` skips the rest of the loop, so removing it forces the rest of the loop to proceed but you don't have the objects to proceed.